### ISSUE TITLE
Save system info, ala sourcecode snapshots

### ISSUE THREAD

**Issue by gar1t** [guildai/guildai#42]
This issue is a place holder to track progress on an upcoming feature for 0.7.

Guild should save system info so that users can compare system differences across runs. 

System info is broadly defined and includes at a minimum:

- Platform information (e.g. information provided by Python's `platform` module)
- Language modules and versions (e.g. output generated by pip freeze)

Refer to https://github.com/guildai/issues/tree/master/42-save-system-info-ala-sourcecode-snapshots for a more detailed treatment of this issue.

**Comment by gar1t** [comment]
This is implemented and pending release in 0.7.0.

**Comment by gar1t** [comment]
Done: Guild saves platform info. E.g.

```
architecture: '64bit '
cpus: 8
processor: x86_64
python_version: 2.7.17 (default, Apr 15 2020, 17:20:14) [GCC 7.5.0]
uname: 'Linux omaha 5.3.0-46-generic #38~18.04.1-Ubuntu SMP Tue Mar 31 04:17:56 UTC
  2020 x86_64 x86_64'
```

As we well as pip freeze output.
